org.eclipse.xtext.testing/src/org/eclipse/xtext/testing/AbstractLanguageServerTest.xtend

```diff
@@ -490,6 +493,18 @@ abstract class AbstractLanguageServerTest implements Endpoint {
 		(List<Either<Command, CodeAction>>)=>void assertCodeActions= null
 	}
 
+    protected def void testSemanticTokensFull((SemanticTokensFullConfiguration)=>void configurator) {
+        val extension configuration = new SemanticTokensFullConfiguration
+        configuration.filePath = 'MyModel.' + fileExtension
+        configurator.apply(configuration)
+        val filePath = initializeContext(configuration).uri
+        val result = languageServer.semanticTokensFull(new SemanticTokensParams=>[
+            textDocument = new TextDocumentIdentifier(filePath)
+        ])
+
+        assertEquals(configuration.expectedText + "\n", result.get.data.toExpectation)
```

Comment:
Randomly added linefeeds at the end of the expected text appear to be very ... random.